File: Backend/customer_account_tracker/src/main/java/com/cts/customer_account_tracker/Service/CustomerService.java
package com.cts.customer_account_tracker.Service;

import java.util.List;

import com.cts.customer_account_tracker.Entity.Customer;
import com.cts.customer_account_tracker.Exceptions.CustomerNotFoundException;

public interface CustomerService {

	public Customer addCustomer(Customer cust);
	public List<Customer> getAllCustomers() throws CustomerNotFoundException;
	public Customer getCustomerById(Long customerId) throws CustomerNotFoundException;
	public String deleteCustomerById(Long customerId);
	public Customer updateCustomer(Long id,Customer cust);
}

File: FrontEnd/customeraccounttrackerapp/src/services/CustomerService.js
import axios from 'axios';

const CUSTOMER_BASE_REST_API_URL = "http://localhost:8081/customers";

class CustomerService{

    addCustomer(customer) {
        return axios.post(CUSTOMER_BASE_REST_API_URL,customer);
    }

    getCustomerById(customerId) {
        return axios.get(CUSTOMER_BASE_REST_API_URL+"/"+customerId);
    }

    getAllCustomers() {
        return axios.get(CUSTOMER_BASE_REST_API_URL);
    }

    updateCustomer(customerId,customer) {
        return axios.put(`${CUSTOMER_BASE_REST_API_URL}/${customerId}`,customer);
    }

    deleteCustomerById(customerId) {
        return axios.delete(CUSTOMER_BASE_REST_API_URL+"/"+customerId);
    }
}

export default new CustomerService();